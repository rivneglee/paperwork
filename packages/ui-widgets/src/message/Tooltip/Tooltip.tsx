import React, { ReactComponentElement, FunctionComponent } from 'react';
import ReactTooltip from 'rc-tooltip';
import 'rc-tooltip/assets/bootstrap.css';

interface Props {
  placement?: 'left' | 'right' | 'top' | 'bottom';
  content: ReactComponentElement<any> | string;
}

const Tooltip: FunctionComponent<Props> = ({
 children,
 placement = 'left',
 content,
}) => (
  <ReactTooltip
    placement={placement}
    overlay={content}
    overlayClassName="pw-tooltip__overlay"
    trigger={[
      'hover',
      'focus',
      'click',
    ]}
  >
    <span>
     {children}
    </span>
  </ReactTooltip>
);

export default Tooltip;
