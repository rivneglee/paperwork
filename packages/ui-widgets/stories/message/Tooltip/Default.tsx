import React from 'react';

import { Tooltip, Icons } from '../../../src';

export default () => (
  <div style={{ background: '#ebeef1', padding: '2.4rem', display: 'flex', justifyContent: 'center' }}>
    <Tooltip content={<span>This is tooltip</span>}>
      <Icons.Form />
    </Tooltip>
  </div>
);
