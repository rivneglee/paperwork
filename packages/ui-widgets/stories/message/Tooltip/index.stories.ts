import { storiesOf } from '@storybook/react';

import Default from './Default';

storiesOf('Tooltip', module)
  .add('default', Default);
