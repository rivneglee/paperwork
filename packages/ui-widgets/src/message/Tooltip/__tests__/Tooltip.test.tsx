import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Tooltip from '../Tooltip';

describe('Tooltip', () => {
  const Trigger = () => <>Trigger</>;

  it('should render for default props', () => {
    const wrapper = shallow(<Tooltip content="This is tooltip"><Trigger /></Tooltip>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
