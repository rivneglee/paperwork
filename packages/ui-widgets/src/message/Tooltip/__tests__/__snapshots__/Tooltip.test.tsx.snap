// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should render for default props 1`] = `
<ForwardRef(Tooltip)
  overlay="This is tooltip"
  overlayClassName="pw-tooltip__overlay"
  placement="left"
  trigger={
    Array [
      "hover",
      "focus",
      "click",
    ]
  }
>
  <span
    className="pw-tooltip__trigger"
  >
    <Trigger />
  </span>
</ForwardRef(Tooltip)>
`;
